opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/batcher.py

```diff
@@ -72,19 +73,20 @@ def process(self, record) -> None:
             get_dict_as_key(aggregator.config),
             record.labels,
         )
+
         batch_value = self._batch_map.get(key)
+
         if batch_value:
-            if batch_value != aggregator:
-                aggregator.take_checkpoint()
-                batch_value.merge(aggregator)
-        else:
-            aggregator.take_checkpoint()
-
-        if self.stateful:
-            # if stateful batcher, create a copy of the aggregator and update
-            # it with the current checkpointed value for long-term storage
-            aggregator = record.aggregator.__class__(
-                config=record.aggregator.config
-            )
-            aggregator.merge(record.aggregator)
+            # Update the stored checkpointed value if exists. The call to merge
+            # here combines only identical records (same key).
+            batch_value.merge(aggregator)
+            return
+
+        # create a copy of the aggregator and update
+        # it with the current checkpointed value for long-term storage
+        aggregator = record.aggregator.__class__(
+            config=record.aggregator.config
+        )
+        aggregator.merge(record.aggregator)
+
         self._batch_map[key] = aggregator
```

Comment:
Was this previously raising an exception (because of unhashable list in dict key) and it doesn't get bubbled up or logged?

Reply:
It was crashing whenever we used HistogramAggregator because it tried to hash the config which is a list.

Reply:
Got it, separate issue to #983